### PR TITLE
Simplify android setup

### PR DESCRIPTION
**Pull request by geof90**
This makes our plugin more compatible with RNPM by making the `CodePush` class itself implement `ReactPackage`. That way, users don't have to modify the auto-generated import statement, or keep a reference to the CodePush instance in their MainActivity.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -312,41 +307,20 @@ After installing the plugin and syncing your Android Studio project with Gradle,
...
-    public class MainActivity extends ReactActivity {
-        ...
-        @Override
-        protected List<ReactPackage> getPackages() {
-            return Arrays.<ReactPackage>asList(
-                ...
-                new CodePushReactPackage() // 2. Remove this line
-                ...
+                new MainReactPackage(), 
+                new CodePush("0dsIDongIcoH0mqAmoR0CYb5FhBZNy1w4Bf-l", this, BuildConfig.DEBUG)
```
Comment (dtivel) [inline README.md]: Should this be a placeholder, like "your deployment key goes here"?

File: README.md [hunk 2]
```diff
@@ -312,41 +307,20 @@ After installing the plugin and syncing your Android Studio project with Gradle,
...
             );
         }
+
         ...
     }
     ```
 
-3. Ensure that the `android.defaultConfig.versionName` property in your `android/app/build.gradle` file is set to a semver compliant value. Note that if the value provided is missing a patch version, the CodePush server will assume it is `0`, i.e. `1.0` will be treated as `1.0.0`.
+2. Ensure that the `android.defaultConfig.versionName` property in your `android/app/build.gradle` file is set to a semver compliant value. Note that if the value provided is missing a patch version, the CodePush server will assume it is `0`, i.e. `1.0` will be treated as `1.0.0`.
```
Comment (dtivel) [inline README.md]: semver-compliant

File: android/app/src/main/java/com/microsoft/codepush/react/CodePush.java [hunk 3]
```diff
@@ -100,6 +101,11 @@ public CodePush(String deploymentKey, Activity mainActivity, boolean isDebugMode
         }
 
         initializeUpdateAfterRestart();
+        if (currentInstance != null) {
+            CodePushUtils.log("More than one CodePush instances has been initialized, please use the non-static method codePush.getBundleUrlInternal() to get the correct bundleURL for a particular instance.");
```
Comment (dtivel) [inline android/app/src/main/java/com/microsoft/codepush/react/CodePush.java]: "More than one CodePush instance has been initialized.  Please use the instance method codePush.getBundleUrlInternal() to get the correct bundleURL for a particular instance."

Comment (dtivel) [inline android/app/src/main/java/com/microsoft/codepush/react/CodePush.java]: Why is one CodePush instance insufficient?  Why not make this a singleton?

File: android/app/src/main/java/com/microsoft/codepush/react/CodePushNotInitializedException.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.microsoft.codepush.react;
+
+public class CodePushNotInitializedException extends RuntimeException {
```
Comment (dtivel) [inline android/app/src/main/java/com/microsoft/codepush/react/CodePushNotInitializedException.java]: final?

File: android/app/src/main/java/com/microsoft/codepush/react/CodePush.java (hunk 3 above)
Comment (geof90) [inline android/app/src/main/java/com/microsoft/codepush/react/CodePush.java]: We know of one user eg. https://github.com/Microsoft/react-native-code-push/pull/255 that is thinking of having multiple ReactActivity in their app. While we don't currently support this scenario now, I wanted to keep the architecture flexible for enabling that in the future.

**Pull request merged**